### PR TITLE
Upstream changes for v0.7.0 release

### PR DESCRIPTION
**Pull request by shubhadeepd**
This release switches all examples to use cloud hosted GPU accelerated LLM and embedding models from [Nvidia API Catalog](https://build.nvidia.com) as default. It also deprecates support to deploy on-prem models using NeMo Inference Framework Container and adds support to deploy accelerated generative AI models across the cloud, data center, and workstation using [latest Nvidia NIM-LLM](https://docs.nvidia.com/nim/large-language-models/latest/introduction.html).

For detailed changes please refer to `CHANGELOG.md` file.

### CODE REVIEW

**Review by nv-pranjald — DISMISSED** [review]

**Review by shubhadeepd — COMMENTED** [review]
Waiting for approval from code owners.

**Review by jliberma — APPROVED** [review]
Looks good, thanks Shubhadeep

**Pull request merged**
